### ISSUE TITLE
Early stopping

### ISSUE THREAD

**Issue by 0liu** [SimonBlanke/Gradient-Free-Optimizers#14]
**Describe the bug**
Optimizers in machine learning libraries, such as scikit-learn, usually come with early stopping ability. Instead of running for a fixed number of iterations, the optimization stops when little improvement could be achieved further, set by parameters like `tol`, `max_iter` and `n_iter_no_change`. This saves a lot computation time. It would be great to have this feature in gradient free optimizers, too. At least, the library can provide an API to stop searching, which can be triggered by an external evaluation logic written by users.

**Additional context**
https://scikit-learn.org/stable/auto_examples/linear_model/plot_sgd_early_stopping.html

**Comment by SimonBlanke** [comment]
The parameter is designed so that you can enable both `tol_`s at the same time. The early stop is triggered if one of both requirements are met.
The feature is ready to go, but I will write some tests to ensure quality before releasing it.

**Comment by SimonBlanke** [comment]
Done!

I just released v0.3.3 with the new early_stopping-feature. Here is an example:
```python
import numpy as np
from gradient_free_optimizers import HillClimbingOptimizer


def parabola_function(para):
    loss = para["x"] * para["x"]
    return -loss


search_space = {"x": np.arange(-10, 10, 0.1)}

opt = HillClimbingOptimizer(search_space)
opt.search(
    parabola_function,
    n_iter=20,
    early_stopping={
        "n_iter_no_change": 5,
        "tol_abs": 0.1,
        "tol_rel": 10,
    },
)
```
